### PR TITLE
[SCB-172] upgrade zipkin libraries to zipkin2

### PR DESCRIPTION
**Pull request by yangbor**
Upgraded zipkin to zipkin2:2.4.2 and zipkin-reporter to zipkin-reporter2:2.2.2.
The zipkin2.Span changed a lot from the old zipkin.Span and the tests are modified accordingly.

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [X] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [X] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [X] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [X] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by WillemJiang** [comment]
@yangbor  ZipkinTracingIntegrationTest is failed, can you double check it?

**Comment by yangbor** [comment]
@WillemJiang The integration test is fixed.

File: integration-tests/pom.xml [hunk 1]
```diff
@@ -54,10 +54,15 @@
         <version>0.6.0-SNAPSHOT</version>
         <type>test-jar</type>
       </dependency>
+      <dependency>
+        <groupId>io.zipkin.zipkin2</groupId>
+        <artifactId>zipkin</artifactId>
+        <version>2.4.2</version>
```
Comment (WillemJiang) [inline integration-tests/pom.xml]: We don't need to specify the version here, the dependency pom can take care of it.

File: integration-tests/test-common/src/test/java/io/servicecomb/tests/tracing/TracingTestBase.java [hunk 2]
```diff
@@ -74,27 +75,33 @@ protected void assertThatSpansReceivedByZipkin(Collection<String> logs, String..
...
-        .filter(span -> span.binaryAnnotations != null)
-        .map(span -> span.binaryAnnotations)
+        .filter(span -> span.tags() != null)
+        .map(span -> span.tags().entrySet())
         .flatMap(Collection::stream)
-        .filter(span -> "call.path".equals(span.key) || "http.path".equals(span.key) || "http.status_code".equals(span.key))
-        .filter(span -> span.value != null)
-        .map(annotation -> new String(annotation.value))
+        .filter(span -> "call.path".equals(span.getKey()) || "http.path".equals(span.getKey()) || "http.status_code"
+            .equals
```
Comment (seanyinx) [inline integration-tests/test-common/src/test/java/io/servicecomb/tests/tracing/TracingTestBase.java]: the format is out of order

File: integration-tests/pom.xml (hunk 1 above)
Comment (yangbor) [reply integration-tests/pom.xml]: This is inside the <dependencyManagement> section so the version is required. The integration-tests/pom.xml does not include the java-chassis-dependencies pom so a separate dependencyManagement is required.

File: integration-tests/test-common/src/test/java/io/servicecomb/tests/tracing/TracingTestBase.java (hunk 2 above)
Comment (yangbor) [reply integration-tests/test-common/src/test/java/io/servicecomb/tests/tracing/TracingTestBase.java]: fixed

**Comment by yangbor** [comment]
@WillemJiang Support for configuring zipkin API version is added.

File: handlers/handler-tracing-zipkin/src/main/java/io/servicecomb/tracing/zipkin/TracingConfiguration.java [hunk 3]
```diff
@@ -19,40 +19,60 @@
...
             DEFAULT_TRACING_COLLECTOR_ADDRESS)
             .trim()
             .replaceAll("/+$", "")
-            .concat(CONFIG_TRACING_COLLECTOR_PATH));
+            .concat(path));
   }
 
   @Bean
   Reporter<Span> zipkinReporter(Sender sender) {
+    if (apiVersion.compareTo(CONFIG_TRACING_COLLECTOR_API_V1) == 0){
```
Comment (WillemJiang) [inline handlers/handler-tracing-zipkin/src/main/java/io/servicecomb/tracing/zipkin/TracingConfiguration.java]: You can use 
CONFIG_TRACING_COLLECTOR_API_V1.equals(apiVersion)
to save some code.

**Comment by WillemJiang** [comment]
@yangbor  Please rebase the code and apply the JIRA ID to your commit log.

**Pull request merged**